code/utilities/helpers/AzureSearchHelper.py

```diff
@@ -60,14 +60,24 @@ def get_vector_store(self):
             ),
         ]
         
+        if env_helper.AZURE_AUTH_TYPE == "keys":
+            search_key = env_helper.AZURE_SEARCH_KEY
```

Comment:
Instead of repeating the full `AzureSearch(...)` block. Can you instead just base the assignment of the `search_key` variable on the condition using [ternary operators](https://book.pythontips.com/en/latest/ternary_operators.html)? E.g:
```python
search_key = env_helper.AZURE_SEARCH_KEY if env_helper.AZURE_AUTH_TYPE == "keys else None
```
Same for the `get_conversation_logger` function below.